plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -1021,61 +1021,67 @@ public function get_coupon_message( $msg_code ) {
 				break;
 			case self::E_WC_COUPON_USAGE_LIMIT_REACHED:
-				$err = __( 'Coupon usage limit has been reached.', 'woocommerce' );
+				/* translators: %s: coupon code */
+				$err = sprintf( esc_html__( 'Usage limit for coupon %s has been reached.', 'woocommerce' ), '"' . esc_html( $this->get_code() ) . '"' );
 				break;
 			case self::E_WC_COUPON_EXPIRED:
-				$err = __( 'This coupon has expired.', 'woocommerce' );
+				/* translators: %s: coupon code */
+				$err = sprintf( esc_html__( 'Coupon %s has expired.', 'woocommerce' ), '"' . esc_html( $this->get_code() ) . '"' );
 				break;
 			case self::E_WC_COUPON_MIN_SPEND_LIMIT_NOT_MET:
-				/* translators: %s: coupon minimum amount */
-				$err = sprintf( __( 'The minimum spend for this coupon is %s.', 'woocommerce' ), wc_price( $this->get_minimum_amount() ) );
+				/* translators: %1$s: coupon code,  %2$s: coupon minimum amount */
+				$err = sprintf( esc_html__( 'The minimum spend for coupon %1$s is %2$s.', 'woocommerce' ), '"' . esc_html( $this->get_code() ) . '"', wc_price( $this->get_minimum_amount() ) );
 				break;
 			case self::E_WC_COUPON_MAX_SPEND_LIMIT_MET:
-				/* translators: %s: coupon maximum amount */
-				$err = sprintf( __( 'The maximum spend for this coupon is %s.', 'woocommerce' ), wc_price( $this->get_maximum_amount() ) );
+				/* translators: %1$s: coupon code, %2$s: coupon maximum amount */
+				$err = sprintf( esc_html__( 'The maximum spend for coupon %1$s is %2$s.', 'woocommerce' ), '"' . esc_html( $this->get_code() ) . '"', wc_price( $this->get_maximum_amount() ) );
 				break;
 			case self::E_WC_COUPON_NOT_APPLICABLE:
-				$err = __( 'Sorry, this coupon is not applicable to your cart contents.', 'woocommerce' );
+				/* translators: %s: coupon code */
+				$err = sprintf( esc_html__( 'Sorry, coupon %s is not applicable to your cart contents.', 'woocommerce' ), '"' . esc_html( $this->get_code() ) . '"' );
 				break;
 			case self::E_WC_COUPON_USAGE_LIMIT_COUPON_STUCK:
 				if ( is_user_logged_in() && wc_get_page_id( 'myaccount' ) > 0 && ! WC()->is_store_api_request() ) {
-					/* translators: %s: myaccount page link. */
-					$err = sprintf( __( 'Coupon usage limit has been reached. If you were using this coupon just now but your order was not complete, you can retry or cancel the order by going to the <a href="%s">my account page</a>.', 'woocommerce' ), wc_get_endpoint_url( 'orders', '', wc_get_page_permalink( 'myaccount' ) ) );
+					/* translators: %1$s: coupon code, %2$s: myaccount page link. */
+					$err = sprintf( wp_kses_data( __( 'Usage limit for coupon "%1$s" has been reached. If you were using this coupon just now but your order was not complete, you can retry or cancel the order by going to the <a href="%2$s">my account page</a>.', 'woocommerce' ) ), esc_html( $this->get_code() ), wc_get_endpoint_url( 'orders', '', wc_get_page_permalink( 'myaccount' ) ) );
```

Comment:
```suggestion
					$err = sprintf( wp_kses_data( __( 'Usage limit for coupon "%1$s" has been reached. If you were using this coupon just now but your order was not complete, you can retry or cancel the order by going to the <a href="%2$s">my account page</a>.', 'woocommerce' ) ), esc_html( $this->get_code() ), esc_attr( wc_get_endpoint_url( 'orders', '', wc_get_page_permalink( 'myaccount' ) ) ) );
```

Reply:
Thanks, fixed in 5fc1b38c35b57af32c81171ee3caf90a409dc0b0.